### PR TITLE
Java: Add models for Commons ToStringBuilder

### PR DESCRIPTION
**Pull request by smowton**
These don't include support for reflectionToString yet, which is coming up in a subsequent PR.

### CODE REVIEW

**Comment by smowton** [comment]
@aschackmull this has been rewritten using the new fluent-method support and CSV specifications.

File: java/ql/src/semmle/code/java/frameworks/apache/Lang.qll [hunk 1]
```diff
@@ -423,3 +423,37 @@ private class ApacheObjectUtilsModel extends SummaryModelCsv {
       ]
   }
 }
+
+private class ApacheToStringBuilderModel extends SummaryModelCsv {
+  override predicate row(string row) {
+    row =
+      [
+        "org.apache.commons.lang3.builder;ToStringBuilder;false;toString;;;Argument[-1];ReturnValue;taint",
+        "org.apache.commons.lang3.builder;ToStringBuilder;false;append;(java.lang.Object);;Argument;ReturnValue;taint",
```
Comment (aschackmull) [inline java/ql/src/semmle/code/java/frameworks/apache/Lang.qll:432]: `Argument` without indices are no longer supported for flow-through.

File: java/ql/src/semmle/code/java/frameworks/apache/Lang.qll [hunk 2]
```diff
@@ -423,3 +423,37 @@ private class ApacheObjectUtilsModel extends SummaryModelCsv {
...
+        "org.apache.commons.lang3.builder;ToStringBuilder;false;append;(java.lang.Object);;Argument;Argument[-1];taint",
+        "org.apache.commons.lang3.builder;ToStringBuilder;false;append;(java.lang.Object[]);;Argument;ReturnValue;taint",
```
Comment (aschackmull) [inline java/ql/src/semmle/code/java/frameworks/apache/Lang.qll:434]: Input should likely be `ArrayElement of Argument[0]` for the various array inputs.

File: java/ql/src/semmle/code/java/frameworks/apache/Lang.qll [hunk 3]
```diff
@@ -423,3 +423,37 @@ private class ApacheObjectUtilsModel extends SummaryModelCsv {
...
+        "org.apache.commons.lang3.builder;ToStringBuilder;false;append;(java.lang.Object[]);;Argument;Argument[-1];taint",
+        "org.apache.commons.lang3.builder;ToStringBuilder;false;append;(java.lang.String,java.lang.Object[]);;Argument;ReturnValue;taint",
```
Comment (aschackmull) [inline java/ql/src/semmle/code/java/frameworks/apache/Lang.qll:436]: Lines like this should be superfluous now, right? With a fluent spec we just need to taint `Argument[-1]`.

**Comment by smowton** [comment]
@aschackmull changes applied; tests expected to fail for the time being.

**Comment by aschackmull** [comment]
> changes applied

It doesn't look like you pushed any changes.

**Comment by smowton** [comment]
Doh, force-pushed (rebase)

**Comment by smowton** [comment]
Test failures expected for now (pending array Content implementation)

**Review by aschackmull — APPROVED** [review]

**Pull request merged**